### PR TITLE
executor: show locked status of accounts

### LINKED ISSUE #37966: SHOW CREATE USER fails to correctly show locked accounts
### 1. Minimal reproduce step (Required)

```sql
CREATE USER 'foo'@'%' IDENTIFIED BY 'foo' ACCOUNT LOCK;
SHOW CREATE USER 'foo'@'%';
```

### 2. What did you expect to see? (Required)

```
sql> CREATE USER 'foo'@'%' IDENTIFIED BY 'foo' ACCOUNT LOCK;
Query OK, 0 rows affected (0.0121 sec)

sql> SHOW CREATE USER 'foo'@'%';
+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| CREATE USER for foo@%                                                                                                                                            |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| CREATE USER 'foo'@'%' IDENTIFIED WITH 'mysql_native_password' AS '*F3A2A51A9B0F2BE2468926B4132313728C250DBF' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT LOCK |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.0017 sec)
```

### 3. What did you see instead (Required)

```
sql> CREATE USER 'foo'@'%' IDENTIFIED BY 'foo' ACCOUNT LOCK;
Query OK, 0 rows affected (0.0121 sec)

sql> SHOW CREATE USER 'foo'@'%';
+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| CREATE USER for foo@%                                                                                                                                            |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| CREATE USER 'foo'@'%' IDENTIFIED WITH 'mysql_native_password' AS '*F3A2A51A9B0F2BE2468926B4132313728C250DBF' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.0017 sec)
```

It should have said `ACCOUNT LOCK` instead.

### 4. What is your TiDB version? (Required)

```
sql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.3.0-alpha-292-gbc0f951d0-dirty
Edition: Community
Git Commit Hash: bc0f951d0a4aff73adaf249383aeddde78649de6
Git Branch: master
UTC Build Time: 2022-09-19 19:08:51
GoVersion: go1.18
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore
1 row in set (0.0009 sec)
```

This is probably related to  #37051

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #37966

Problem Summary: `SHOW CREATE USER` fails to correctly show if a user is locked or not.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by CbcWestwolf** [comment]
LGTM

**Review by CbcWestwolf — APPROVED** [review]

File: executor/showtest/show_test.go [hunk 1]
```diff
@@ -1095,6 +1095,12 @@ func TestShowCreateUser(t *testing.T) {
 	// Compare only the start of the output as the salt changes every time.
 	rows = tk.MustQuery("SHOW CREATE USER 'sock2'@'%'")
 	require.Equal(t, "CREATE USER 'sock2'@'%' IDENTIFIED WITH 'auth_socket' AS 'sock3' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK", rows.Rows()[0][0].(string))
+
+	// Test ACCOUNT LOCK/UNLOCK
+	tk.MustExec("CREATE USER 'lockness'@'%' IDENTIFIED BY 'monster' ACCOUNT LOCK")
+	rows = tk.MustQuery("SHOW CREATE USER 'lockness'@'%'")
+	require.Equal(t, "CREATE USER 'lockness'@'%' IDENTIFIED WITH 'mysql_native_password' AS '*BC05309E7FE12AFD4EBB9FFE7E488A6320F12FF3' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT LOCK", rows.Rows()[0][0].(string))
+
```
Comment (CbcWestwolf) [inline executor/showtest/show_test.go:1103]:
```suggestion
```

Comment (CbcWestwolf) [reply executor/showtest/show_test.go:1103]: Please remove this line to pass the check.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**